hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -207,7 +207,7 @@ static const ColorFormatInfo kelvin_color_format_map[66] = {
          {GL_ONE, GL_ONE, GL_ONE, GL_RED}},
     [NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8Y8] =
         {2, true, GL_RG8, GL_RG, GL_UNSIGNED_BYTE,
-         {GL_GREEN, GL_GREEN, GL_GREEN, GL_RED}},
+         {GL_RED, GL_RED, GL_RED, GL_GREEN}},
```

Comment:
This component swizzling stuff can be a little confusing when reading, especially if you don't know the byte order of the texture data. To other reviewers, when using the texture format associated with `NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8Y8`, each pixel in the texture has `Y8` as the first byte (address-wise, that is) of the pixel, then `A8` as the second byte of the pixel.

We use the GL format `GL_RG` and GL internal format `GL_RG8` format here with type `GL_UNSIGNED_BYTE`. This maps the first byte of a texel to the red component and the second byte to the green component. At texture bind time we then apply texture component swizzling such that when the fragment shader reads a sampled texel's `.r`, `.g`, or `.b` component, it will get the texel's "red" component (which in the texture data provided by the title is assumed to be the `Y8` channel), and accordingly when the shader reads a sampled texel's `.a` component, it will get the value from the "green" component of the texel (the `A8` channel in provided texture data).